pkg/reconciler/route/route.go

```diff
@@ -227,6 +227,10 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	traffic, err := c.configureTraffic(ctx, r, clusterLocalServiceNames)
 	if traffic == nil || err != nil {
 		// Traffic targets aren't ready, no need to configure child resources.
+		// However, still need to update observed generation because otherwise
+		// route's ready status won't be propagated to service (Service's RoutesReady
+		// will stay "Unknown")
+		r.Status.ObservedGeneration = r.Generation
```

Comment:
There are a number of other places that the Route reconciler could fail both before and after this that could get us into a similar Route=False/Service=Unknown state.

Why not just the update to ObservedGeneration further down in this function up to the beginning of the function since we want to update this for all successes and failures?

Reply:
Are we okay with updating ObservedGeneration in all failure cases in route reconciler?

Reply:
Let me take a deeper look here as there might be a cleaner way. If all of the failed cases result in the Route's Ready=False then yes.